packages/near-membrane-base/src/membrane.ts

```diff
@@ -1052,20 +1050,32 @@ export function createMembraneMarshall(
         ) {
             delete lazyState[key];
             let safeDesc;
-            foreignCallableGetOwnPropertyDescriptor(
-                getTransferablePointer(target),
-                key,
-                (configurable, enumerable, writable, valuePointer, getPointer, setPointer) => {
-                    safeDesc = createDescriptorFromMeta(
-                        configurable,
-                        enumerable,
-                        writable,
-                        valuePointer,
-                        getPointer,
-                        setPointer
-                    );
-                }
-            );
+            const foreignTargetPointer = getTransferablePointer(target);
+            const activity = instrumentation
```

Comment:
Currently, @locker/sandbox passes a dummy instrumentation object to near-membrane if the container does not provide one. Maybe we should also ensure a dummy instrumentation object in near-membrane so that we can avoid all existence checks?

Reply:
Actually, by inlining the instrumentation calls, these checks are always executed. Are we favoring extra checks than a wrapper call?

Reply:
checks are easy and JITs will fast path them. Dummy calls are not great as they have to also inline the call because a function call is an unknown and so the engine has to JIT and inline the dummy call. It's easy to just always be a undefined and what not. Not in favor of dummy objects. No object or `undefined` is great.

Reply:
Engines have a fixed amount of inline function caching. Minification also helps v8 since it judges things based on source size (yes large comments can bust its cache)